drivers/hmis/lib/form_data/static/external_form_submission_review.json

```diff
@@ -15,7 +15,16 @@
       "text": "Mark reviewed",
       "mapping": {
         "field_name": "reviewed"
-      }
+      },
+      "enable_when": [
+        {
+          "local_constant": "$reviewed",
```

Comment:
Still wrapping my head around the order of ops here (if the local constant is even the right approach) so that everything is backwards compatible. If we merge this PR first and the frontend changes after, the reviewed checkbox will be disabled for every submission while that local constant is not yet present.

Maybe we need to merge 3 separate PRs:
1. to hmis-frontend release branch, adding this local constant to the context
2. merge this PR to hmis-warehouse pit-external-forms branch
3. to hmis-frontend pit-external-forms branch, with schema updates and UX for bulk review

Reply:
This does seem to work, but it also just feels like the "static form" paradigm just doesn't align super neatly with what we want to do. We already have some custom log for setting the [initial values](https://github.com/greenriver/hmis-frontend/blob/6d7f14fe7a862dfa30671b30b7b2e3505d3260bc/src/modules/projects/components/tables/ProjectExternalSubmissionsTable.tsx#L124-L130) on the form, which is somewhat non-standard, and now we also want to hide or disabling the "reviewed" option based on whether it is has been reviewed.

Imo we should only use a "static form" when it simplifies development. I don't _think_ that's really the case here, so maybe we should just switch the frontend to use a custom form. I think that would simplify things and make it easier to maintain. (It would also let us improve the UI around the "mark reviewed" vs "mark spam" later on, for example making them buttons..)

What do you think?


replacing static form dialog with something like:
```
<CommonDialog>
  <DialogTitle></DialogTitle>
  <DialogContent>
    <DynamicView />
    <TextInput label='Reviewer Notes' multiline minRows={2} />
    <LabeledCheckbox label='Mark as Reviewed' />
    <LabeledCheckbox label='Mark as Spam' />
  </DialogContent>
  <DialogActions></DialogActions>
</CommonDialog>
```

Reply:
Sure I think that makes sense. In that case, maybe we should also get rid of the static form dialog's `beforeFormComponent` argument, which is only used here.

Reply:
In terms of timing, should we make this change in the frontend against the release branch (not the pit branch)? That way we can also deprecate the external form static form role enum on the release branch, and then later, either as part of the pit branch or on another release, we can remove it entirely.

Reply:
That sounds reasonable to me!